src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,80 @@
+  const dispatch = useAppDispatch();
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    if (currentTodo) {
+      const loadUser = async () => {
+        try {
+          const currentUser = await getUser(currentTodo.userId);
+
+          setUser(currentUser);
+        } finally {
+          setIsLoading(false);
+        }
+      };
+
+      loadUser();
+    }
+  }, [currentTodo]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${currentTodo?.id}`}
+            </div>
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
-
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() =>
+                dispatch(currentTodoSlice.actions.clearCurrentTodo())
+              }
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {currentTodo?.title}
+            </p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+            <p className="block" data-cy="modal-user">
+              {/* For not completed */}
+              {!currentTodo?.completed && (
+                <strong className="has-text-danger">Planned</strong>
+              )}
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+              {/* For completed */}
+              {currentTodo?.completed && (
+                <strong className="has-text-success">Done</strong>
+              )}
+              {' by '}
+              <a href="mailto:Sincere@april.biz">{user?.name}</a>
```

Comment:
The email address in the anchor tag is hardcoded as 'Sincere@april.biz'. It should be dynamically set using the `user` object, like `user?.email`, to ensure it reflects the actual user's email.